### ISSUE TITLE
Moving `experimental_clipping_planes` out of experimental

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#4466]
## 🚀 Feature
This PR is an idea for one way we could go to make @brisvag's clipping planes a mainline feature 

Proposed API:

```python
Viewer.clipping_planes: ClippingPlaneList
Layer.clipped: bool = True
```

`Viewer.clipping_planes` are a set of planes which clip the data in the scene, defined in scene coordinates
the `clipped` attribute on each layer controls whether a specific layer is clipped by the clipping planes, true by default

## Motivation

This experimental feature has been experimental for a while and adds real value, we should figure out what to do with it

## Alternatives

This API does not support subsets of layers being clipped differently - we may want to consider something more flexible which allows for this possibility (but would necessarily be more complex)

**Comment by jni** [comment]
Could there be both viewer clipping planes and layer clipping planes? (Layers inherit the viewer ones but can have additional ones.)

We should also think about the `viewer.canvases` migration and whether this should get tackled at the same time. (See #4148.)

**Comment by brisvag** [comment]
I like this API as a starting point: it does what most people would want to do and exposes it in an intuitive way. We will need to find a place in the GUI to put controls for these planes...

> Could there be both viewer clipping planes and layer clipping planes? (Layers inherit the viewer ones but can have additional ones.)

I'm afraid it will cause a proliferation of plane models scattered everywhere and a confusion on where the objects "are". I would like to bring up #3148 in order to not lose the thoughts we had over there :)

**Comment by Chris-N-K** [comment]
May I make a suggestion for the GUI placement?
I think it might be a good idea to either add them like 3D display to the bottom left buttons with a right click pop-up for control or to the same layer control drop-down menu like plane mode as they are somewhat comparable.

**Comment by brisvag** [comment]
Yes, you got it. I would say they are completely unrelated. They *could* be related, but when discussing with @jni and @andy-sweet (I can't find the conversation, sorry), we agreed that using the dims `thickness` to create axis-aligned clipping planes for the `displayed_dims` was more confusing than helpful. Happy to revisit the conversation if anyone disagrees!

**Comment by kephale** [comment]
> > Could there be both viewer clipping planes and layer clipping planes? (Layers inherit the viewer ones but can have additional ones.)
> 
> I'm afraid it will cause a proliferation of plane models scattered everywhere and a confusion on where the objects "are". I would like to bring up #3148 in order to not lose the thoughts we had over there :)

I do like the idea of masks that follow the scene graph hierarchy via LayerGroups (discussed in #3148). The GLSL generation in vispy's PlanesClipper would need some work to add new mask geometries and support for composing masks. 

If a fancier masking/clipping system is interesting to folks, then that might impact whether this moves out of experimental and how seriously we take the GUI implementation.
